src/_P039_Thermosensors.ino

```diff
@@ -1789,7 +1811,7 @@ void write8BitRegister(uint8_t l_CS_pin_no, uint8_t l_address, uint8_t value)
 
 void write16BitRegister(uint8_t l_CS_pin_no, uint8_t l_address, uint16_t value)
 {
-  uint8_t l_messageBuffer[3] = {l_address, (uint8_t) (value >> 8), (uint8_t) (value)};
+  uint8_t l_messageBuffer[3] = {l_address, static_cast<uint8_t> (value >> 8), static_cast<uint8_t> (value)};
```

Comment:
`static_cast<uint8_t> (value)`  -> `static_cast<uint8_t> (value & 0xFF)`

Reply:
OK - that's fail safe programming. I would end up with the following:

``uint8_t l_messageBuffer[3] = {l_address, static_cast<uint8_t> ((value >> 8) & 0xFFu), static_cast<uint8_t> (value & 0xFFu)};`` 

-> part of my next push

Reply:
`0xFFu` is not needed (not even valid I guess) as HEX is always unsigned.
Shifted to the right always has the most significant bits set to 0. 8 bit shifted on a 16 bit value can be cast safely to an 8 bit value.

Reply:
Solved in latest push d908816